xml_converter/src/xml_converter.cpp

```diff
@@ -144,20 +152,29 @@ void process_data(
 
     // This is a special output path used for burrito internal use that splits
     // the guildpoint protobins by map id.
-    string output_split_guildpoint_dir) {
+    string output_split_guildpoint_dir,
+    bool allow_duplicates) {
     // All of the loaded pois and categories
     vector<Parseable*> parsed_pois;
     map<string, Category> marker_categories;
+    set<string> top_level_categories;
+    vector<string> duplicate_categories;
 
     // Read in all the xml taco markerpacks
     auto begin = chrono::high_resolution_clock::now();
     for (size_t i = 0; i < input_taco_paths.size(); i++) {
         cout << "Loading taco pack " << input_taco_paths[i] << endl;
 
-        read_taco_directory(
+        set<string> category_names = read_taco_directory(
             input_taco_paths[i],
             &marker_categories,
             &parsed_pois);
+        for (string category_name : category_names) {
```

Comment:
The tooling to check if a category is a duplicate or not should probably be broken out into its own function.

Reply:
I agree. I'll spin out this function. I imagine this function would take in a pointer to the set and vector. Is this specific enough it should be in this file or should I include it in the `string_helper.hpp` functions

Reply:
I made a more generic function that combines two sets and reports the duplicates between them.